### ISSUE TITLE
When db:dump command runs out of memory, it still returns status code 0

### ISSUE THREAD

**Issue by hostep** [platformsh/platformsh-cli#644]
Hi

This is based on my experience with the [magento-cloud cli tool](http://devdocs.magento.com/guides/v2.2/cloud/reference/cli-ref-topic.html), which seems to be a clone of this tool, but I can't find a public repo to report this to, so if this is inappropriate here, feel free to close.

I had a couple of occurrences where running the `db:dump` command which uses `mysqldump` runs out of memory somehow, with the error:
```
magento-cloud.phar db:dump -f "dbdump-develop-2017-11-11-11-13-10-326028.sql.gz" -d "./tools/dbdumps" -z -e "develop"
Creating gzipped SQL dump file: /Users/me/Projects/someproject/tools/dbdumps/dbdump-develop-2017-11-11-11-13-10-326028.sql.gz
mysqldump: Got errno 12 on write
```

In our particular script this should halt all the following commands, since the sql file is corrupted by this error.
But I think (unverified) that the `db:dump` command still returns status code `0` which makes our script think that everything went just fine. It would be great if `db:dump` could catch this problem and then return something else then status code `0`.
It's hard to reproduce this problem, most of the times `mysqldump` runs perfectly fine, but in some rare cases it runs against this out of memory problem, so it's hard to try to reproduce this, hence why I'm not 100% sure if the status code isn't `0`, because I wasn't able to verify this yet.

Thanks!

**Comment by pjcdawkins** [comment]
I actually suspect this might happen from the `--gzip` option, which means you'll receive the exit code from the pipe, not from the mysqldump command.

https://github.com/platformsh/platformsh-cli/blob/298279e2d1d2efdadf3a14520ecead911abdf111/src/Command/Db/DbDumpCommand.php#L164

That looks a bug, and maybe it can be resolved with a `set -o pipefail`.

**Comment by hostep** [comment]
Nice, sounds like this is probably the correct solution!

**Comment by pjcdawkins** [comment]
Fixed in [v3.22.3](https://github.com/platformsh/platformsh-cli/releases/tag/v3.22.3)
